### PR TITLE
Support for txn forwarding to Espresso sequencer

### PR DESCRIPTION
**Pull request by nomaxg**
Add configuration options + logic to forward transactions to an instance of HotShot.

### CODE REVIEW

File: arbitrum/backend.go [hunk 1]
```diff
@@ -67,10 +70,57 @@ func (b *Backend) ChainDb() ethdb.Database {
 	return b.chainDb
 }
 
+type Transaction struct {
+	Vm      uint64 `json:"vm"`
+	Payload []int  `json:"payload"`
```
Comment (sveitser) [inline arbitrum/backend.go]: Wondering why this isn't `[]uint8` instead.

File: arbitrum/backend.go [hunk 2]
```diff
@@ -67,10 +70,57 @@ func (b *Backend) ChainDb() ethdb.Database {
...
+		return err
+	}
+	request.Header.Set("Content-Type", "application/json")
+	client := &http.Client{}
+	response, err := client.Do(request)
+	if err != nil {
+		return err
+	}
+	if response.StatusCode != 200 {
+		return err
```
Comment (sveitser) [inline arbitrum/backend.go]: This will return `nil`. Is this intentional?

File: arbitrum/backend.go [hunk 3]
```diff
@@ -67,10 +70,57 @@ func (b *Backend) ChainDb() ethdb.Database {
...
+	request, err := http.NewRequest("POST", b.config.HotShotUrl, bytes.NewBuffer(marshalled))
+	if err != nil {
+		return err
+	}
+	request.Header.Set("Content-Type", "application/json")
+	client := &http.Client{}
+	response, err := client.Do(request)
+	if err != nil {
+		return err
+	}
```
Comment (sveitser) [inline arbitrum/backend.go:118]: I think we should close response bodies if `err != nil` (?) https://pkg.go.dev/net/http
```suggestion
	}
	defer response.Body.Close()
```

File: arbitrum/config.go [hunk 4]
```diff
@@ -60,6 +65,9 @@ func ConfigAddOptions(prefix string, f *flag.FlagSet) {
 	arbDebug := DefaultConfig.ArbDebug
 	f.Uint64(prefix+".arbdebug.block-range-bound", arbDebug.BlockRangeBound, "bounds the number of blocks arbdebug calls may return")
 	f.Uint64(prefix+".arbdebug.timeout-queue-bound", arbDebug.TimeoutQueueBound, "bounds the length of timeout queues arbdebug calls may return")
+	f.Bool(prefix+".espresso", DefaultConfig.Espresso, "if true, l2 transactions will be fetched from espresso sequencer")
```
Comment (sveitser) [inline arbitrum/config.go]: At the moment this change causes transactions to be forwarded.

```suggestion
	f.Bool(prefix+".espresso", DefaultConfig.Espresso, "if true, the espresso sequencer is used for sequencing and DA")
```

**Review by sveitser — APPROVED** [review]
LGTM. Some small stuff.

**Review by ImJeremyHe — APPROVED** [review]

File: arbitrum/config.go [hunk 5]
```diff
@@ -37,6 +37,11 @@ type Config struct {
 	ClassicRedirect        string        `koanf:"classic-redirect"`
 	ClassicRedirectTimeout time.Duration `koanf:"classic-redirect-timeout"`
 	MaxRecreateStateDepth  int64         `koanf:"max-recreate-state-depth"`
+
+	// Espresso specific flags
+	Espresso          bool   `koanf:"espresso"`
+	HotShotUrl        string `koanf:"hotshot-url"`
```
Comment (ImJeremyHe) [inline arbitrum/config.go:43]: This is a bit confusing for me. As `HotShot` is our protocol of consensus and it doesn't seem necessary for people knowing anything about `HotShot` in the integration. Maybe `EspressoUrl` is more appropriate?

**Review by ImJeremyHe — APPROVED** [review]

File: arbitrum/config.go (hunk 5 above)
Comment (nomaxg) [reply arbitrum/config.go:43]: Yeah I think you have a point, although we seem to have developed a convention for referring to "HotShot" for all things consensus (we have variable names like hotshot_contract, hotshot_url, etc. sprinkled all over the place). This is definitely something to think about as we start making more client facing tools like SDKs. I personally think its OK to refer to HotShot directly for now.

**Pull request merged**